mesh/nurbs.hpp

```diff
@@ -120,71 +195,144 @@ class NURBSPatch
    int nd; // Number of knots in flattened structure
    int ls; // Number of variables per knot in flattened structure
    int sd; // Stride for data access
+
+   /// Flattens the B-NET in direction @a dir, producing a 1D net.
+   // Returns the number of variables per knot in flattened structure.
    int SetLoopDirection(int dir);
+
+   /** @brief Access function for the effectively 1D flattened net, where @a i
+       is a knot index, and @a j is an index of a variable per knot. */
    inline       double &slice(int i, int j);
    inline const double &slice(int i, int j) const;
 
+   /// Copy constructor
    NURBSPatch(NURBSPatch *parent, int dir, int Order, int NCP);
+
+   /// Deletes own data, takes data from @a np, and deletes np.
    void swap(NURBSPatch *np);
-   void init(int dim_);
+
+   /// Sets dimensions and allocates data, based on knotvectors.
+   /// @a dim is the physical dimension plus 1.
+   void init(int dim);
 
 public:
+   /// Copy constructor
    NURBSPatch(const NURBSPatch &orig);
+
+   /// Constructor using data read from stream @a input.
    NURBSPatch(std::istream &input);
+
+   /// Constructor for a 2D patch.
+   // dim is physical dimension plus 1
```

Comment:
There are several instances in which a comment starts with `//`. Is this compatible with doxygen?

Reply:
Good catch. I removed these, except for comment lines that are not intended to be in doxygen.